gov/politeia/piclient/piclient.go

```diff
@@ -95,15 +95,15 @@ func RetrieveAllProposals(client *http.Client, APIRootPath, URLParams string) (
 
 // RetrieveProposalByToken returns a single proposal identified by the token
 // hash provided if it exists. Data returned is queried from Politeia API.
-func RetrieveProposalByToken(client *http.Client, APIRootPath, token string) (*pitypes.ProposalInfo, error) {
+func RetrieveProposalByToken(client *http.Client, APIRootPath, token string) (*pitypes.Proposal, error) {
 	// Constructs the full proposal's URl and fetch is data.
 	proposalRoute := APIRootPath + DropURLRegex(piapi.RouteProposalDetails, token)
 	data, err := HandleGetRequests(client, proposalRoute)
 	if err != nil {
 		return nil, fmt.Errorf("retrieving %s proposal details failed: %v", token, err)
 	}
 
-	var proposal pitypes.ProposalInfo
+	var proposal pitypes.Proposal
 	err = json.Unmarshal(data, &proposal)
 	if err != nil {
 		return nil, err
```

Comment:
Also  need to check if `proposal.Data` is `nil`. `json.Unmarshal` will not error, but `proposal.Data` will be `nil` if Pi returns:

```json
{
    "proposal": null
}
```

https://play.golang.org/p/0nLHmiJaWny

Reply:
Similarly should check any other pointer types in `ProposalInfo` that could cause a panic later.

Reply:
Since it unmarshals without error, I just meant to check `if proposal.Data == nil { return nil, [error] }`